vllm/compilation/collective_fusion.py

```diff
@@ -0,0 +1,418 @@
+import operator
+from typing import List, Tuple
+
+import torch
+import torch.fx as fx
+from torch._inductor.pattern_matcher import (Match, PatternMatcherPass,
+                                             fwd_only, register_replacement)
+
+import vllm._custom_ops as ops
+import vllm.envs as envs
+from vllm.compilation.inductor_pass import InductorPass
+from vllm.compilation.utils import (find_auto_fn, find_fn, find_getitem,
+                                    last_node_in_match)
+from vllm.distributed import (tensor_model_parallel_all_gather,
+                              tensor_model_parallel_all_reduce)
+from vllm.distributed.parallel_state import (
+    get_tensor_model_parallel_rank, get_tensor_model_parallel_world_size,
+    get_tp_group)
+from vllm.logger import init_logger
+from vllm.utils import direct_register_custom_op
+
+logger = init_logger(__name__)
+
+use_flux = False
+if envs.VLLM_USE_FLUX:
+    try:
+        import flux
+        use_flux = True
+        logger.info("USING FLUX")
+    except ImportError:
+        use_flux = False
+
+# TODO: factor out somehow
+TP_GROUP_NAME = "tp:0"
```

Comment:
Do you know what will happen if there are multiple TP groups (i.e. in a pipeline parallel setting)?

Reply:
I've factored out the group names so that they will be extracted from the matched code and reused in the replacement.